authors/apps/authentication/views.py

```diff
@@ -73,3 +75,48 @@ def update(self, request, *args, **kwargs):
 
         return Response(serializer.data, status=status.HTTP_200_OK)
 
+class SocialSignUp(CreateAPIView):
+    permission_classes = (AllowAny,)
+    renderer_classes = (UserJSONRenderer,)
+    serializer_class = UserSerializer
+
+    def create(self, request,  *args, **kwargs):
+        # Get access token and create user or authenticate
+
+        provider = request.data['provider']
```

Comment:
Handle error if this is not provided. You can use a serializer.

Reply:
Thanks for noticing that. I'll work on it.